lib/src/over_react_redux/over_react_redux.dart

```diff
@@ -0,0 +1,393 @@
+///               ConnectedBigCounter()(
+///                 Dom.h4()('ConnectedCounter Store1'),
+///                 ConnectedCounter()(),
+///               ),
+///             ),
+///           ),
+///         ),
+///       ), querySelector('#content')
+///     );
+///
+/// - [pure] if `true` (default), connect performs several equality checks that are used to avoid unnecessary
+/// calls to [mapStateToProps], [mapDispatchToProps], [mergeProps], and ultimately to [render]. These include
+/// [areStatesEqual], [areOwnPropsEqual], [areStatePropsEqual], and [areMergedPropsEqual].
+/// While the defaults are probably appropriate 99% of the time, you may wish to override them with custom
+/// implementations for performance or other reasons.
+///
+/// - [forwardRef] if `true`, the `ref` prop provided to the connected component will be return the wrapped component.
+///
+/// For more info see: https://react-redux.js.org/api/connect#connect
+UiFactory<TProps> Function(UiFactory<TProps>) connect<TReduxState, TProps extends UiProps>({
+  Map Function(TReduxState state) mapStateToProps,
+  Map Function(TReduxState state, TProps ownProps) mapStateToPropsWithOwnProps,
+  Map Function(dynamic Function(dynamic) dispatch) mapDispatchToProps,
+  Map Function(dynamic Function(dynamic) dispatch, TProps ownProps) mapDispatchToPropsWithOwnProps,
+  Map Function(TProps stateProps, TProps dispatchProps, TProps ownProps) mergeProps,
+  bool Function(TReduxState next, TReduxState prev) areStatesEqual,
+  bool Function(TProps next, TProps prev) areOwnPropsEqual,
+  bool Function(TProps next, TProps prev) areStatePropsEqual,
+  bool Function(TProps next, TProps prev) areMergedPropsEqual,
+  Context context,
+  bool pure: true,
+  bool forwardRef: false,
+}) {
+  areStatesEqual ??= _defaultEquality;
+  areOwnPropsEqual ??= _shallowMapEquality;
+  areStatePropsEqual ??= _shallowMapEquality;
+  areMergedPropsEqual ??= _shallowMapEquality;
+
+  UiFactory<TProps> wrapWithConnect(UiFactory<TProps> factory) {
+    final dartComponentClass = (factory().componentFactory as ReactDartComponentFactoryProxy).reactClass;
+
+    JsMap handleMapStateToProps(ReactInteropValue jsState) {
+      final state = unwrapInteropValue(jsState);
+
+      var result = mapStateToProps(state);
+      // Pull out the underlying backing map to avoid copying where possible
+      if (result is UiProps) result = (result as UiProps).props;
+      return jsBackingMapOrJsCopy(result);
+    }
+
+    JsMap handleMapStateToPropsWithOwnProps(ReactInteropValue jsState, JsMap jsOwnProps) {
+      final state = unwrapInteropValue(jsState);
+      final ownProps = factory(new JsBackedMap.backedBy(jsOwnProps));
+
+      var result = mapStateToPropsWithOwnProps(state, ownProps);
+      // Pull out the underlying backing map to avoid copying where possible
+      if (result is UiProps) result = (result as UiProps).props;
+      return jsBackingMapOrJsCopy(result);
```

Comment:
Might be worth splitting out this logic since it's used multiple times:
```dart
// Pull out the underlying backing map to avoid copying where possible
JsMap jsMapFromProps(Map props) => 
    jsBackingMapOrJsCopy(props is UiProps ? result.props : props)
```